src/compass/utils/validate_cslc.py

```diff
@@ -0,0 +1,144 @@
+    return parser.parse_args()
+
+
+def compare_cslc_products(file_1, file_2):
+    '''
+    Compare a reference and a newly generated
+    (i.e., secondary) CSLC product
+
+    Parameters
+    ----------
+    file1: str
+        File path to reference CSLC product (golden dataset)
+    file2: str
+        File path to generated CSLC product to use for comparison
+    '''
+
+    # Check if file paths exits
+    if not os.path.exists(file_1):
+        print(f'ERROR reference CSLC product not found: {file_1}')
+        return
+
+    if not os.path.exists(file_2):
+        print(f'ERROR secondary CSLC product not found: {file_2}')
+        return
+
+    # Extract some info from reference/secondary CSLC products
+    dataset_1 = gdal.Open(file_1, gdal.GA_ReadOnly)
+    geotransform_1 = dataset_1.GetGeoTransform()
+    nbands_1 = dataset_1.RasterCount
+
+    dataset_2 = gdal.Open(file_2, gdal.GA_ReadOnly)
+    geotransform_2 = dataset_2.GetGeoTransform()
+    nbands_2 = dataset_2.RasterCount
+
+    # Compare number of bands
+    print('Comparing CSLC number of bands ...')
+    if not nbands_1 == nbands_2:
+        print(f'ERROR Number of bands in reference CSLC {nbands_1} differs'
+              f'from number of bands {nbands_2} in secondary CSLC')
+        return
+
+    print('Comparing geo transform arrays ...')
+    if not np.array_equal(geotransform_1, geotransform_2):
+        print(f'ERROR Reference geo transform array {geotransform_1} differs'
+              f'from secondary CSLC geo transform array {geotransform_2}')
+        return
+
+    # Compare amplitude of reference and generated CSLC products
+    slc_1 = dataset_1.GetRasterBand(1).ReadAsArray()
+    slc_2 = dataset_2.GetRasterBand(1).ReadAsArray()
+
+    diff_amp = np.abs(slc_1) - np.abs(slc_2)
+    diff_pha = np.angle(slc_1 * np.conj(slc_2))
+
+    print('Check max amplitude difference between CSLC products is < 1.0e-12')
+    assert np.nanmax(diff_amp) < 1.0e-12
+    print('Check max phase difference between CSLC products is < 1.0e-12')
+    assert np.nanmax(diff_pha) < 1.0e-12
+    return
+
+
+def compare_cslc_metadata(file_1, file_2):
+    '''
+    Compare reference and generated CSLC metadata
+    '''
+
+    # Check if metadata files exists
+    if not os.path.exists(file_1):
+        print(f'ERROR reference CSLC metadata not found: {file_1}')
+        return
+
+    if not os.path.exists(file_2):
+        print(f'ERROR CSLC metadata not found: {file_2}')
+        return
+
+    # Load metadata
+    with open(file_1, 'r') as f:
+        metadata_1 = json.load(f)
+
+    with open(file_2, 'r') as f:
+        metadata_2 = json.load(f)
+
+    print('Compare number of metadata keys')
+    if not len(metadata_1.keys()) == len(metadata_2.keys()):
+        print('ERROR different number of metadata keys')
+        return
+
+    # Intersect metadata keys
+    set_1_m_2 = set(metadata_1.keys()) - set(metadata_2.keys())
+    if len(set_1_m_2) > 0:
```

Comment:
```suggestion
    if set_ref_minus_sec:
```
Python sets, IIRC built-in containers in general, return `False` empty. Also rename for clarity.

Reply:
Not sure this comment is doable. I get the following error with your suggestion:

```
Traceback (most recent call last):
  File "/u/aurora0/vbrancat/codes/validate_cslc.py", line 141, in <module>
    compare_cslc_metadata(cmd.ref_metadata, cmd.sec_metadata)
  File "/u/aurora0/vbrancat/codes/validate_cslc.py", line 116, in compare_cslc_metadata
    if set_ref_minus_sec > 0:
TypeError: '>' not supported between instances of 'set' and 'int'

```